src/urllib3/connection.py

```diff
@@ -447,17 +447,16 @@ def connect(self) -> None:
         self._connecting_to_proxy = bool(self.proxy)
 
         sock: Union[socket.socket, "ssl.SSLSocket"]
-        sock = self._new_conn()
+        self.sock = sock = self._new_conn()
```

Comment:
this socket is closed here if `conn.connect()` fails: https://github.com/urllib3/urllib3/blob/main/src/urllib3/connectionpool.py#L809-L810

Reply:
fix is pretty much what cpython does: https://github.com/python/cpython/blob/dd69f734218ac5d3a551227069ac53ee09b0cd3e/Lib/http/client.py#L1444-L1455

Reply:
this weird Exception/close responsibility is forced on urrlib3 by cpython http.client.HTTPConnection: https://github.com/python/cpython/blob/dd69f734218ac5d3a551227069ac53ee09b0cd3e/Lib/urllib/request.py#L1346-L1355